File: pset6/sentimental-credit/credit.py
while True:
    number = input("Number: ")
    digits = len(number)
    try:
        number = int(number)
        break
    except ValueError:
        pass

if digits not in [13, 15, 16]:
    print("INVALID")
else:
    total = 0
    number = str(number)
    if digits % 2 == 0:
        for i in range(0, digits, 2):
            total += int(number[i]) * 2 // 10 + int(number[i]) * 2 % 10
        for i in range(1, digits, 2):
            total += int(number[i])
    else:
        for i in range(0, digits, 2):
            total += int(number[i])
        for i in range(1, digits, 2):
            total += int(number[i]) * 2 // 10 + int(number[i]) * 2 % 10
    if total % 10 == 0:
        if number[:2] in ['34', '37'] and digits == 15:
            print("AMEX")
        elif number[:2] in ['51', '52', '53', '54', '55'] and digits == 16:
            print("MASTERCARD")
        elif number[0] == '4' and (digits == 13 or digits == 16):
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long number = get_long("Number: ");
    int count = 0;
    long k = number;
    long i = number;
    if (k == 0)
    {
        count = 1;
    }
    while (k != 0)
    {
        k /= 10;
        count++;
    }

    while (i >= 100)
    {
        i /= 10;
    }

    if (count == 13 || count == 15 || count == 16)
    {
        int sum = 0;
        while (number != 0)
        {
            sum += number % 10;
            number /=10;
            if ((number % 10) * 2 >= 10)
            {
                sum +=  ((number % 10) * 2) / 10 + ((number % 10) * 2) % 10;
            }
            else
            {
                sum += (number % 10) * 2;
            }
            number /=10;
        }
        if (sum % 10 == 0)
        {
            if (count == 15 && (i == 34 || i == 37))
            {
                printf("AMEX\n");
            }
            else if (count == 16 && (i >= 51 && i <= 55))
            {
                printf("MASTERCARD\n");
            }
            else if ((count == 13 || count == 16) && i / 10 == 4)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}
